http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -1885,7 +1885,7 @@ private void writeFinalNettyResponse(MutableHttpResponse<?> message, HttpRequest
             if (!isHttp2) {
                 if (!nettyHeaders.contains(HttpHeaderNames.CONNECTION)) {
                     boolean expectKeepAlive = nettyResponse.protocolVersion().isKeepAliveDefault() || request.getHeaders().isKeepAlive();
-                    if (!expectKeepAlive || httpStatus.getCode() > 499) {
+                    if ((!expectKeepAlive || httpStatus.getCode() > 499) && !serverConfiguration.isKeepAliveOnServerError()) {
```

Comment:
This seems wrong because if the setting was set to true it would would always be keep alive, even on non server error states.

It should probably be `if (expectKeepAlive || httpStatus.getCode() < 500 || serverConfiguration.isKeepAliveOnServerError()) { KEEP_ALIVE } else { CLOSE }`

Reply:
updated. @jameskleeh agree this one is more clear.